src/tactic/sanity_check.lean

```diff
@@ -3,24 +3,46 @@ Copyright (c) 2019 Floris van Doorn. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Floris van Doorn
 -/
-
 import tactic.core
-
 /-!
-# sanity_check command
- This file defines the `#sanity_check` and `#sanity_check_mathlib` commands, to spot common mistakes in the current file or in all of mathlib, respectively.
+  # sanity_check command
+  This file defines the following user commands to spot common mistakes in the code.
+  * `#sanity_check`: check all declarations in the current file
+  * `#sanity_check_mathlib`: check all declarations in mathlib (so excluding core or other projects,
+    and also excluding the current file)
+  * `#sanity_check_all`: check all declarations in the environment (the current file and all
+    imported files)
+
+  Currently this will check for
+  1. unused arguments in declarations,
+  2. whether a declaration is incorrectly marked as a def/lemma,
+  3. whether a namespace is duplicated in the name of a declaration
+  4. whether ≥/> is used in the declaration
+
+  You can append a `-` to any command (e.g. `#sanity_check_mathlib-`) to omit the slow tests (4).
 
- Currently this will check for unused arguments in declarations and whether a declaration is incorrectly marked as a def/lemma.
+  You can customize the performed checks like this:
+  ```
+  meta def my_check (d : declaration) : tactic (option string) :=
+  return $ if d.to_name = `foo then some "gotcha!" else none
+  run_cmd sanity_check tt [(my_check, "found nothing", "found something")] >>= trace
+  ```
 
- ## Tags
- sanity check, sanity_check, cleanup, command, tactic
+  ## Tags
+  sanity check, sanity_check, cleanup, command, tactic
 -/
 
 universe variable u
 open expr tactic native
 
 reserve notation `#sanity_check`
 reserve notation `#sanity_check_mathlib`
+reserve notation `#sanity_check_all`
+
+@[user_attribute]
+meta def sanity_skip_attr : user_attribute :=
```

Comment:
Proposal for the future: make this attribute take a list of names of checks to skip. E.g., if I have a reason to add an unused argument to a definition (see some type tags), I may still want to run all other checks.